### ISSUE TITLE
Re-Used Notification When Uploading Same File Twice Or More at the same time

### ISSUE THREAD

**Issue by hendrawd** [gotev/android-upload-service#213]
### Version info
* Android Upload Service version: 3.0.3
* Android version and API version: any
* HTTP stack (e.g. HurlStack or OkHttpStack): HurlStack

### What did you expect?
Seperate notification

### What happened instead?
Re-used notification

### Steps to reproduce (if applicable):
I have the same file uploaded twice or more at the same time.
The notification from that different uploads will use the same notification.

### Library initialization code:
```java
UploadService.NAMESPACE = BuildConfig.APPLICATION_ID;
```

### Request code:
```java
final MultipartUploadRequest multipartUploadRequest = new MultipartUploadRequest(this, mCurrentPhotoPath, url)
.setNotificationConfig(getNotificationConfig(mCurrentPhotoPath))
.setMaxRetries(0);
multipartUploadRequest.addFileToUpload(uriToUpload.getPath(), "file");
multipartUploadRequest.addParameter("some param key", "value");
multipartUploadRequest.addHeader("some cookie key", "value");
multipartUploadRequest.startUpload();
```

#### Where have you added the request code?
- [x] Activity
- [ ] Service
- [ ] Other class (add additional info about it)

### Additional info
Basically the code in the `Request code` section above, called twice or more  in a short time with the same file.

**Comment by gotev** [comment]
Made a fix and added to 3.1 dev version. You can try it from JitPack. This includes also fix for #214 

```groovy
repositories {
    maven { url 'https://jitpack.io' }
}
```

```groovy
compile 'com.github.gotev:android-upload-service:c5d3b2940f'
```
Let me know if the issue can be considered fixed. Thank you!

**Comment by hendrawd** [comment]
Do you think it is a good solution? If i as a user actually want to upload the same file multiple times with different parameters it will fail.
Example case if i have photo sharing application that use your library,
☞ User have a photo
☞ User upload it with parameter `user_id = 1`
☞ User upload the same photo with parameter `user_id = 2` right away
☞ The second upload will fail

**Comment by gotev** [comment]
I think it's a good solution because you are not supposed to re-use the same UploadRequest object twice and just change a parameter and then call the `start` method again. You should create a new UploadRequest for every operation, so basically just encapsulate the UploadRequest object creation in a function and pass the needed parameters (as shown in the Recipes snippets)

**Comment by hendrawd** [comment]
Yes, actually that is what i did in my code. I always create new `MultipartUploadRequest` object every time. But if the file path is same, notification will be re-used

**Comment by hendrawd** [comment]
@gotev is the request same with my example?
Make sure you set the upload id here
new MultipartUploadRequest(this, **mCurrentPhotoPath**, url)

**Comment by gotev** [comment]
if you do like that, of course you will get that behaviour, because every upload request should have a unique id

**Comment by hendrawd** [comment]
There is a reason why i set the ID directly. I can't get the ID directly from `MultipartUploadRequest` until i call `multipartUploadRequest.startUpload();`. But i need the ID when setting up `MultipartUploadRequest` to be used as a canceller, because user can cancel the upload anytime. I suppose i need to generate unique ID by myself to achieve it.

**Comment by gotev** [comment]
The library simply generates an UUID4 string, which you can generate and achieve the same thing

**Comment by hendrawd** [comment]
Yes, i know. Sorry i wasn't really clear from my last post. I edited your library to use notification action button #181  because of my requirement. The button just simply a cancel action. 

I can create NotificationAction by:
```java
NotificationAction actionCancel = new NotificationAction(R.drawable.ic_close, getString(R.string.notification_active_action_cancel_upload), UploadCanceller.getPendingIntent(this, uploadId));
```

Then i set it to NotificationConfig
```java
private UploadNotificationConfig getNotificationConfig(String uploadId) {
    return new UploadNotificationConfig().addAction(actionCancel);//i set other configs too
}
```

Then i set it to MultipartUploadRequest
```java
.setNotificationConfig(getNotificationConfig(mCurrentPhotoPath));
```

I can't get the id before i call
```java
String idNeeded = multipartUploadRequest.startUpload();
```

But i must set the canceller with ID before call startUpload, that's why i create the ID directly

**Comment by gotev** [comment]
@hendrawd you have explained your scenario well even in the previous post. What I suggest you to do in your case is simply: 
```java
String uploadId = UUID.randomUUID().toString();

MultipartUploadRequest req = new MultipartUploadRequest(this, uploadId, url);
```
The `id` which you will get from:
```java
String id = req.startUpload();
```
in this case is the same `uploadId` you have passed in the constructor, so you know the `id` even before the `startUpload` method invocation, because you generated it. Scenario like yours are exactly the reasor why I've implemented also the constructor in which you can pass your own uploadId.

Some reference:
[MultipartUploadRequest constructor](http://gotev.github.io/android-upload-service/javadoc/net/gotev/uploadservice/MultipartUploadRequest.html#MultipartUploadRequest-android.content.Context-java.lang.String-java.lang.String) | [startUpload method](http://gotev.github.io/android-upload-service/javadoc/net/gotev/uploadservice/UploadRequest.html#startUpload--)
